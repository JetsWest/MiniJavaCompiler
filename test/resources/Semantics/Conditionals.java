class SomeClass {
    public static void main(String[] args) {
        System.out.println(new A());
    }
}

class Object { }

class A {
    int[] arr;
    boolean bool1;
    boolean bool2;
    int num1;
    int num2;
    Object obj;

    public int foo() {
        if (true) { }
        else { }

        if (false) { }
        else { }

        if (bool1) { }
        else { }

        if (bool1 && bool2) { }
        else { }

        if (!bool1) { }
        else { }

        if (num1 < num2) { }
        else { }

        if (arr.length) { }
        else { }

        if (arr[0]) { }
        else { }

        if (arr) { }
        else { }

        if (num1) { }
        else { }

        if (num1 + num2) { }
        else { }

        if (num1 - num2) { }
        else { }

        if (num1 * num2) { }
        else { }

        if (1) { }
        else { }

        if (obj) { }
        else { }

        return 0;
    }
}