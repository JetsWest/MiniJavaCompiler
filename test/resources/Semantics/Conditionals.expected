(Line 42) If statement must contain a boolean type.
(Line 45) If statement must contain a boolean type.
(Line 48) If statement must contain a boolean type.
(Line 51) If statement must contain a boolean type.
(Line 54) If statement must contain a boolean type.
(Line 57) If statement must contain a boolean type.
(Line 60) If statement must contain a boolean type.